pkg/antctl/raw/supportbundle/command.go

```diff
@@ -641,9 +654,149 @@ func processResults(resultMap map[string]error, dir string) error {
+	agentInfoList, err := antreaClientset.CrdV1beta1().AntreaAgentInfos().List(context.TODO(), metav1.ListOptions{ResourceVersion: "0"})
+	if err != nil {
+		return err
+	}
+
+	agentInfoMap := map[string]v1beta1.AntreaAgentInfo{}
+	for _, agentInfo := range agentInfoList.Items {
+		agentInfoMap[agentInfo.Name] = agentInfo
+	}
+
+	pods, err := k8sClient.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{
+		ResourceVersion: "0",
+		LabelSelector:   "app=antrea",
+	})
+	if err != nil {
+		return err
+	}
+
+	failedNodesMap := make(map[string]struct{})
+	for _, node := range failedNodes {
+		failedNodesMap[node] = struct{}{}
+	}
+
+	var errors []error
+
+	for _, pod := range pods.Items {
+
+		tmpDir, err := afero.TempDir(defaultFS, "", "bundle_tmp_")
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		defer defaultFS.RemoveAll(tmpDir)
+
+		if pod.Labels["component"] == "antrea-controller" && isControllerFail {
+			controllerInfo, err := antreaClientset.CrdV1beta1().AntreaControllerInfos().Get(context.TODO(), "antrea-controller", metav1.GetOptions{})
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+			data, err := yaml.Marshal(controllerInfo)
+			if err != nil {
+				errors = append(errors, err)
+				continue
+			}
+			err = afero.WriteFile(defaultFS, filepath.Join(tmpDir, "controllerinfo"), data, 0644)
+			errors = append(errors, err)
+
+			err = downloadPodLogs(k8sClient, "controller", pod.Namespace, pod.Name, []string{"antrea-controller"}, dir, tmpDir)
+			errors = append(errors, err)
+
+		}
+
+		if _, exist := failedNodesMap[pod.Spec.NodeName]; !exist {
+			continue
+		}
+
+		if pod.Labels["component"] == "antrea-agent" {
+			if agentInfo, ok := agentInfoMap[pod.Spec.NodeName]; ok {
+				data, err := yaml.Marshal(agentInfo)
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+				err = afero.WriteFile(defaultFS, filepath.Join(tmpDir, "agentinfo"), data, 0644)
+				errors = append(errors, err)
+			}
+
+			err = downloadPodLogs(k8sClient, "agent_"+pod.Spec.NodeName, pod.Namespace, pod.Name, []string{"antrea-agent", "antrea-ovs", "install-cni"}, dir, tmpDir)
+			errors = append(errors, err)
+
+		}
+	}
+	return utilerror.NewAggregate(errors)
+}
+
+func downloadPodLogs(k8sClient kubernetes.Interface, comp string, namespace string, podName string, containers []string, dir string, tmpDir string) error {
+	var errors []error
+
+	for _, containerName := range containers {
+		containerDirName := containerName
+		if strings.HasPrefix(containerName, "antrea-") {
+			containerDirName = strings.ReplaceAll(containerName, "antrea-", "")
+		}
+
+		podLogDir := filepath.Join(tmpDir, "logs", containerDirName)
+		err := os.MkdirAll(podLogDir, 0755)
+		if err != nil {
+			return err
+		}
+
+		fileName := filepath.Join(podLogDir, containerName+".log")
+		f, err := os.Create(fileName)
```

Comment:
I didn't find where this f gets closed. I think in general it's better to call f.Close() explicitly rather than relying on GC to run file object finalizer to close it. Otherwise, if there are many failed Pods, antctl may open a lot of files without closing it and end up in FD limit error.